pkg/hubble/observer/local_observer.go

```diff
@@ -430,6 +430,7 @@ func (r *eventsReader) Next(ctx context.Context) (*observerpb.GetFlowsResponse,
 				},
 			}, nil
 		case *flowpb.AgentEvent:
+			r.eventCount++
```

Comment:
Is there a reason not to `r.eventCount++` for `*flowpb.LostEvent` as well? If not, I'd suggest moving `r.eventCount++` above the switch case.

Reply:
I think it's a question of what semantics we want to provide. We do exclude `LostEvents` from `until`/`since`, because it's timestamp is not monotonic.

If a user writes `hubble observe -t policy-verdict --last 12` and there are 4 arbitrary lost events in between the most recent flow of the ring buffer and the twelfth policy verdict, then there are two potential options here:

1. We return 12 policy verdict events and 4 lost events (current implementation)
2. We return 8 policy verdict events and 4 lost events (result of doing `r.eventCount++` on lost events)

Option 2 seems more intuitive at first, but the lost events might have not been policy verdict events, so arguably they should not have been counted. If there are 20 lost events followed by 50 policy verdict events, then option 2 would never return any policy verdict events at all.

It's not obvious to me what the correct solution here is. We might even want to consider the user to opt-in to lost events?

Reply:
I see, good point. Option 1 makes more sense to me so lost events are treated as a special case.

> We might even want to consider the user to opt-in to lost events?

How valuable is this information to a user? Maybe what should be done is simply count the number of lost events to report as a metric.